solr/solrj/src/test/org/apache/solr/client/solrj/impl/LBHttp2SolrClientTest.java

```diff
@@ -108,8 +108,13 @@ public void testAsyncWithFailures() {
           } catch (TimeoutException | ExecutionException e) {
             fail(iterMessage + " Response ended in failure: " + e);
           }
-          if (j == 0) {
-            // The first endpoint gives an exception, so it retries.
+          if (i == 0) {
```

Comment:
[+1] Appreciate the explanatory comment (though it looks like 'tidy' messed up your formatting)!